climpred/smoothing.py

```diff
@@ -0,0 +1,157 @@
+                Gonzalez, V. Kharin, et al. “A Verification Framework for
+                Interannual-to-Decadal Predictions Experiments.” Climate
+                Dynamics 40, no. 1–2 (January 1, 2013): 245–72.
+                https://doi.org/10/f4jjvf.
+
+        Args:
+            ds (xr.object): input. xr.DataArray prefered.
+            boxsize (tuple): target grid.
+                             Defaults to `(5, 5)` (see Goddard et al. 2013).
+
+        Returns:
+            ds_smoothed (xr.object): boxsize-regridded input
+
+        """
+        if isinstance(ds, xr.Dataset):
+            was_dataset = True
+            ds = ds.to_array().squeeze()
+        else:
+            was_dataset = False
+        ds_out = xe.util.grid_global(*boxsize)
+        regridder = xe.Regridder(ds, ds_out, 'bilinear')
+        res = regridder(ds)
+
+        if was_dataset:
+
+            def _variable_dataarray_to_dataset(da):
+                """Return dataarray from dataset.to_darray() to dataset."""
+                data_vars = [i for i in da['variable'].values]
+                print(data_vars)
+                da_list = []
+                for i in data_vars:
+                    a = da.sel(variable=i).squeeze()
+                    a.name = i
+                    del a['variable']
+                    da_list.append(a)
+                res = xr.merge(da_list)
+                return res
+
+            res = _variable_dataarray_to_dataset(res)
+        return res
+
+
+except:
+    print('xesmf couldnt be loaded.')
+
+
+def spatial_smoothing_xrcoarsen(ds, coarsen_dict=None, how='mean'):
+    """Apply spatial smoothing by regridding to `boxsize` grid.
+
+    Reference:
+      * Goddard, L., A. Kumar, A. Solomon, D. Smith, G. Boer, P.
+            Gonzalez, V. Kharin, et al. “A Verification Framework for
+            Interannual-to-Decadal Predictions Experiments.” Climate
+            Dynamics 40, no. 1–2 (January 1, 2013): 245–72.
+            https://doi.org/10/f4jjvf.
+
+    Args:
+        ds (xr.object): input. xr.DataArray prefered.
+        coarsen_dict (dict): coarsen spatial latitudes.
+        how (str): aggregation type for coarsening. default: 'mean'
+
+    Returns:
+        ds_smoothed (xr.object): boxsize-regridded input
+
+    """
+    if coarsen_dict is None:
+        # guess spatial dims
+        spatial_dims_to_smooth = ds.dims.drop(['time', 'lead', 'member', 'init'])
+        # write coarsen to dict to coarsen similar to 5x5 degree
+        pass  # not implemented
+    # check whether coarsen dims are possible
+    for dim in coarsen_dict:
+        if dim not in ds.dims:
+            raise ValueError(dim, 'not in ds')
+        else:
+            if ds[dim].size % coarsen_dict[dim] != 0:
+                raise ValueError(
+                    coarsen_dict[dim], 'does not divide', ds[dim].size, 'in', dim
+                )
+    ds_out = ds.coarsen(coarsen_dict)
+    if how == 'mean':
```

Comment:
is there any way to generalize this more? to use the str in as `ds.coarsen(coarsen_dict).unstr(how)`?

Reply:
I think what you want is
`ds_out = getattr(ds, how)()`